### ISSUE TITLE
[BUG] Boolean value

### ISSUE THREAD

**Issue by ElGigi** [atlasphp/Atlas.Transit#9]
Hi,

In MySQL the boolean value is similar to TINYINT (0: false / 1: true).

In my Domain class, i give in constructor parameters the bool type value, it's work fine when i select a domain. The integer is convert to boolean.

But when i persist the Domain class:
1. The value taken and given to the mapper is boolean value
2. The Mapper found that the value is different (boolean <> int)
3. The ORM execute the UPDATE
4. The ORM throw an exception because PDO return no updated entry (because boolean converted to integer)

Ronan

**Comment by pmjones** [comment]
Yes, I have run into this myself elsewhere. The solution I used was to add a TableEvent::beforeUpdate() to cast the value to integer, a la:

```php
function beforeUpdate($table, $row)
{
    $row->foo = (int) $row->foo;
}
```

(That happens before the array diff gets calculated, so the int <=> int comparison works correctly.)

It was a little more involved than that, as I looped through the $table column definitions and cast them automatically if they were TINYINT, but the principle is the same.

**Comment by pmjones** [comment]
Perhaps the thing to do would be, on the way back to the Record, for the Handlers to cast the domain value to the same type as the original Record value.
